src/evict/evict_lru.c

```diff
@@ -1521,7 +1521,18 @@ __wt_cache_eviction_worker(WT_SESSION_IMPL *session, int busy, int pct_full)
 				return (0);
 		}
 
-		/* Wait for the queue to re-populate before trying again. */
+		/*
+		 * Wait 1/10th second for the queue to re-populate before trying
+		 * again; if we've waited for a second, decrement the counter.
+		 * The decrement is defensive (otherwise the loop might never
+		 * terminate), and there have been bugs in other parts of the
+		 * code causing the wrong thread to infinitely loop here.
+		 */
+		if (++wait_count > 10) {
+			wait_count = 0;
+			if (--count == 0)
+				return (0);
+		}
```

Comment:
I understand the motivation, but it seems like this will allow the cache to grow in situations where it is bounded today.  It will grow slowly (1 operation per second per thread), but we're going over the allocated cache size.

How much testing of this have you done?  Was there a particular workload that prompted this part of the change?